src/AuthManager.ts

```diff
@@ -50,6 +49,74 @@ export class AuthManager {
         return { newCodeVerifier, newCodeChallenge };
     };
 
+    private async refreshAccessToken(): Promise<string> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                const refreshToken: string | null = localStorage.getItem('refresh_token');
+                if (!refreshToken) {
+                    throw new Error('No refresh token found');
+                }
+                const decodedRefreshToken = JSON.parse(atob(refreshToken.split('.')[1]));
+                if (decodedRefreshToken) {
+                    const currentTime = Date.now() / 1000;
+                    if (decodedRefreshToken.exp < currentTime) {
+                        throw new Error('Refresh token expired');
+                    }
+                }
+                await fetch(`${this.authServer}auth/refresh`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify({
+                        refresh_token: refreshToken,
+                    }),
+                })
+                    .then((response) => {
+                        if (response.status !== 200) {
+                            throw new Error('Failed to refresh the token');
+                        }
+                        return response.json();
+                    })
+                    .then((exchangeJson) => {
+                        localStorage.setItem('refresh_token', exchangeJson.refresh_token);
+                        localStorage.setItem('access_token', exchangeJson.access_token);
+                        resolve(exchangeJson.access_token);
+                    })
+                    .catch((error) => {
+                        reject(error);
+                    });
+            } catch (error) {
+                reject(error);
+            }
+        });
+    }
```

Comment:
missing empty line between methods, do we have eslint-supaapps installed here?

Reply:
no linting & no prettier Should i add them in this PR?